crates/nu-command/src/filesystem/glob.rs

```diff
@@ -138,6 +162,12 @@ impl Command for Glob {
                 },
             )
             .flatten()
+            .filter(|entry| match () {
+                () if no_dirs && entry.file_type().is_dir() => false,
+                () if no_files && entry.file_type().is_file() => false,
+                () if no_symlinks && entry.file_type().is_symlink() => false,
+                () => true,
```

Comment:
I wonder if getting the filetype once is better? maybe like
```
            .filter(|entry| {
                let ft = entry.file_type();
                no_dirs && !ft.is_dir()
                    || no_files && !ft.is_file()
                    || no_symlinks && !ft.is_symlink()
            })
```

Reply:
I think it would, but considering that most invocations of `glob` don't pass any filter and thus don't access the file type, I decided not to.

Also, I think you're looking at an old version of the code? That version had a bug, where if you didn't pass any filters, none would match. I'm glad you guys had tests for `glob` and I added a few more.

Reply:
I could be looking at an old one for sure, there have been several in the past few minutes. LOL.

More tests is good. However we make this, we need to make sure we're not getting file_type for each check because it seems like that would be a drag on the performance of glob. I'm not sure what the right answer is though.

Reply:
Yeah there were indeed a few lol

Looking at the source, it seems like it defers to walkdir's `file_path()`, [which is just a property access](https://docs.rs/walkdir/latest/src/walkdir/dent.rs.html#158-160). So it should be fine. But I did add the caching just because it looks cleaner.